### PR TITLE
implement login validator

### PR DESCRIPTION
**Pull request by dmitry-rachko-idf**

### CODE REVIEW

File: packages/presentation/build/.last_build_id [hunk 1]
```diff
@@ -0,0 +1 @@
+683a223b2249c2d86526835f64555848
```
Comment (andrei-lastochkin-idf) [inline packages/presentation/build/.last_build_id:1]: I guess this folder should be ignored

File: packages/presentation/build/683a223b2249c2d86526835f64555848/.filecache [hunk 2]
```diff
@@ -0,0 +1 @@
+{"version":2,"files":[{"path":"/Users/dmitrij/develop/FlutterApplications/authprodapp/packages/presentation/lib/l10n/app_ru.arb","hash":"cf8f666e5ebdac4850713ae048b36c97"},{"path":"/Users/dmitrij/develop/FlutterApplications/authprodapp/packages/presentation/lib/l10n/app_en.arb","hash":"e1e1aac8ca6a956fa37b3958ce65c794"},{"path":"/Users/dmitrij/develop/FlutterApplications/authprodapp/packages/presentation/.dart_tool/flutter_gen/gen_l10n/app_localizations_en.dart","hash":"fe116a9bd354efb1de442c8baf3d5b08"},{"path":"/Users/dmitrij/develop/FlutterApplications/authprodapp/packages/presentation/l10n.yaml","hash":"02908c2b98049a8c54871515602b0a64"},{"path":"/Users/dmitrij/develop/FlutterApplications/authprodapp/packages/presentation/.dart_tool/flutter_gen/gen_l10n/app_localizations.dart","hash":"fdcdf21c5c32e21ae7334b7c2031c232"},{"path":"/Users/dmitrij/develop/FlutterApplications/authprodapp/packages/presentation/.dart_tool/flutter_gen/gen_l10n/app_localizations_ru.dart","hash":"a025d13c894ce04e9bb061f8b243eb85"},{"path":"/Users/dmitrij/develop/flutter/packages/flutter_tools/lib/src/build_system/targets/localizations.dart","hash":"2e40a6f704dbf5e274a7cb59b6d7c0c9"}]}
```
Comment (andrei-lastochkin-idf) [inline packages/presentation/build/683a223b2249c2d86526835f64555848/.filecache:1]: -//-

File: packages/presentation/build/683a223b2249c2d86526835f64555848/gen_l10n_inputs_and_outputs.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{"inputs":["/Users/dmitrij/develop/FlutterApplications/authprodapp/packages/presentation/lib/l10n/app_en.arb","/Users/dmitrij/develop/FlutterApplications/authprodapp/packages/presentation/lib/l10n/app_ru.arb"],"outputs":["/Users/dmitrij/develop/FlutterApplications/authprodapp/packages/presentation/.dart_tool/flutter_gen/gen_l10n/app_localizations_en.dart","/Users/dmitrij/develop/FlutterApplications/authprodapp/packages/presentation/.dart_tool/flutter_gen/gen_l10n/app_localizations_ru.dart","/Users/dmitrij/develop/FlutterApplications/authprodapp/packages/presentation/.dart_tool/flutter_gen/gen_l10n/app_localizations.dart"]}
```
Comment (andrei-lastochkin-idf) [inline packages/presentation/build/683a223b2249c2d86526835f64555848/gen_l10n_inputs_and_outputs.json:1]: -//-

File: packages/presentation/build/683a223b2249c2d86526835f64555848/gen_localizations.d [hunk 4]
```diff
@@ -0,0 +1 @@
+ /Users/dmitrij/develop/FlutterApplications/authprodapp/packages/presentation/.dart_tool/flutter_gen/gen_l10n/app_localizations_en.dart /Users/dmitrij/develop/FlutterApplications/authprodapp/packages/presentation/.dart_tool/flutter_gen/gen_l10n/app_localizations_ru.dart /Users/dmitrij/develop/FlutterApplications/authprodapp/packages/presentation/.dart_tool/flutter_gen/gen_l10n/app_localizations.dart:  /Users/dmitrij/develop/FlutterApplications/authprodapp/packages/presentation/l10n.yaml /Users/dmitrij/develop/FlutterApplications/authprodapp/packages/presentation/lib/l10n/app_en.arb /Users/dmitrij/develop/FlutterApplications/authprodapp/packages/presentation/lib/l10n/app_ru.arb
```
Comment (andrei-lastochkin-idf) [inline packages/presentation/build/683a223b2249c2d86526835f64555848/gen_localizations.d:1]: -//-

File: packages/presentation/build/683a223b2249c2d86526835f64555848/outputs.json [hunk 5]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (andrei-lastochkin-idf) [inline packages/presentation/build/683a223b2249c2d86526835f64555848/outputs.json:1]: -//-

File: packages/presentation/lib/screens/login_page/login_page.dart [hunk 6]
```diff
@@ -1,9 +1,12 @@
...
 import 'package:presentation/screens/app/app_page.dart';
 import 'package:presentation/screens/login_page/login_bloc.dart';
-import 'package:presentation/screens/login_page/login_data.dart';
 import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:presentation/screens/login_page/switch_error.dart';
+import 'package:presentation/widget/global_text_form_field.dart';
+import 'package:presentation/widget/password_text_form_field.dart';
 
 import '../../base/bloc_state.dart';
 import '../../base/stream_platform_stack_content.dart';
```
Comment (andrei-lastochkin-idf) [inline packages/presentation/lib/screens/login_page/login_page.dart:12]: it's a good practice to provide full path

File: packages/presentation/lib/screens/login_page/login_page.dart [hunk 7]
```diff
@@ -39,106 +40,86 @@ class _LoginPageState extends BlocState<LoginPage, LoginBloc> {
...
-                  decoration: const InputDecoration(
-                    border: OutlineInputBorder(
-                      borderSide: BorderSide(color: Colors.green, width: 2.0),
-                    ),
+              Padding(
+                padding: const EdgeInsets.only(top: 8),
+                child: AppTextFormField(
+                  globalKey: bloc.loginFieldKey,
+                  focusNode: bloc.loginFocusNode,
+                  validator: (value) => SwitchError().errorToMessage(
```
Comment (andrei-lastochkin-idf) [inline packages/presentation/lib/screens/login_page/login_page.dart:60]: provide mapper via di

File: packages/presentation/lib/screens/login_page/login_page.dart [hunk 8]
```diff
@@ -39,106 +40,86 @@ class _LoginPageState extends BlocState<LoginPage, LoginBloc> {
...
-                  initialValue: screenData.passwordInput,
-                  validator: (value) => value == null || value.trim().isEmpty
-                      ? 'Invalid password'
-                      : null,
+              Padding(
+                padding: const EdgeInsets.only(top: 8),
+                child: PasswordTextFormField(
+                  globalKey: bloc.passwordFieldKey,
+                  focusNode: bloc.passwordFocusNode,
+                  validator: (value) => SwitchError().errorToMessage(
```
Comment (andrei-lastochkin-idf) [inline packages/presentation/lib/screens/login_page/login_page.dart:81]: -//-

File: packages/presentation/lib/screens/login_page/login_page.dart [hunk 9]
```diff
@@ -39,106 +40,86 @@ class _LoginPageState extends BlocState<LoginPage, LoginBloc> {
...
+                    child: MaterialButton(
+                      height: 56,
+                      minWidth: double.infinity,
+                      onPressed: bloc.login,
+                      child: Text(
+                        AppLocalizations.of(context)!.buttonLogin,
+                        style: const TextStyle(
+                          fontWeight: FontWeight.bold,
+                          color: Colors.white,
+                        ),
```
Comment (andrei-lastochkin-idf) [inline packages/presentation/lib/screens/login_page/login_page.dart:114]: move text styles to separate class and get them as a field/method that returns this style

**Review by andrei-lastochkin-idf — CHANGES_REQUESTED** [review]

File: packages/presentation/lib/screens/login_page/login_page.dart (hunk 6 above)
Comment (dmitry-rachko-idf) [reply packages/presentation/lib/screens/login_page/login_page.dart:12]: I will fix it in Main arch pr

**Pull request closed without merging**